src/Autocomplete.tsx

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable @typescript-eslint/indent */
+import * as React from 'react';
+import { useState } from 'react';
+import { Person } from './types/Person';
+import debounce from 'lodash.debounce';
+
+interface AutocompleteProps {
+  people: Person[];
+  delay?: number;
+  onSelected: (person: Person | null) => void;
+}
+
+export const Autocomplete: React.FC<AutocompleteProps> = ({
+  people,
+  delay = 300,
+  onSelected,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [suggestions, setSuggestions] = useState<Person[]>([]);
+  const [isDropDownActive, setIsDropDownActive] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const filterSuggestions = debounce((value: string) => {
+    const results = value
+      ? people.filter(person =>
+          person.name.toLowerCase().includes(value.toLowerCase()),
+        )
+      : people;
+
+    setSuggestions(results);
+    setIsDropDownActive(true);
+  }, delay);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setInputValue(value);
+    filterSuggestions(value);
+
+    if (selectedPerson) {
+      setSelectedPerson(null);
+      onSelected(null);
+    }
```

Comment:
The check for `selectedPerson` before setting it to null and calling `onSelected(null)` is a good practice to avoid unnecessary state updates. However, consider moving this logic outside of the `handleInputChange` function if it's not directly related to input changes, to keep the function focused on its primary responsibility.